cassandra/connection.py

```diff
@@ -673,6 +677,13 @@ def process_result(result):
 
         self.send_msg(query, request_id, process_result)
 
+    @property
+    def is_idle(self):
+        return self.in_flight == 0 and not self.msg_received
```

Comment:
Hmm, I'm not completely sure that the heartbeat should only be issued if self.in_flight is 0.  For example, if a switch started dropping packets, an operation could hang (with in_flight > 0) for quite some time.  The user would get an OperationTimedOut() for an operation, but the connection wouldn't necessarily be defuncted right away.  The heartbeat timeing out could help to defunct the connection more quickly in that case if the in_flight check is removed.